pkgs/misc/fastly/default.nix

```
@@ -0,0 +1,53 @@
+{ lib, fetchFromGitHub, installShellFiles, buildGoModule }:
+
+buildGoModule rec {
+  pname = "fastly";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "fastly";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "sha256-pMe9hsjxJ4zyzupUWWU2tc5sJb9xhIIdG/Xjh2d7XtY=";
+    # The git commit is part of the `fastly version` original output;
+    # leave that output the same in nixpkgs.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD | head -c7 > $out/COMMIT
```

```suggestion
      git rev-parse HEAD --short > $out/COMMIT
```

should also work?

just invalidate the sha to check but don't be surprised if it says the exact same sha since it should be the exact same output

---

Yes, updated to use `git rev-parse --short HEAD`. It was giving me more characters than expected, but I just learnt the width of `--short` is adaptative depending on the number of commits...

---

ah, cool. even if they get enough commits at some point it should be the same with `--short` since that's what they use upstream

https://github.com/fastly/cli/blob/main/Makefile#L6